src/sentry/models/apitoken.py

```diff
@@ -28,10 +29,98 @@ def default_expiration():
     return timezone.now() + DEFAULT_EXPIRATION
 
 
-def generate_token():
+def generate_token(token_type: AuthTokenType | str | None = AuthTokenType.__empty__) -> str:
+    if token_type:
+        return f"{token_type}{secrets.token_hex(nbytes=32)}"
+
     return secrets.token_hex(nbytes=32)
 
 
+class PlaintextSecretAlreadyRead(Exception):
+    def __init__(
+        self,
+        message="the secret you are trying to read is read-once and cannot be accessed directly again",
+    ):
+        super().__init__(message)
+
+
+class NotSupported(Exception):
+    def __init__(self, message="the method you called is not supported by this token type"):
+        super().__init__(message)
+
+
+class ApiTokenManager(ControlOutboxProducingManager):
+    def create(self, *args, **kwargs):
+        token_type: AuthTokenType | None = kwargs.get("token_type", None)
+
+        # Typically the .create() method is called with `refresh_token=None` as an
+        # argument when we specifically do not want a refresh_token.
+        #
+        # But if it is not None or not specified, we should generate a token since
+        # that is the expected behavior... the refresh_token field on ApiToken has
+        # a default of generate_token()
+        #
+        # TODO(mdtro): All of these if/else statements will be cleaned up at a later time
+        # to use a match statment on the AuthTokenType. Move each of the various token type
+        # create calls one at a time.
+        if "refresh_token" in kwargs:
+            plaintext_refresh_token = kwargs["refresh_token"]
+        else:
+            plaintext_refresh_token = generate_token()
+
+        if token_type == AuthTokenType.USER:
+            plaintext_token = generate_token(token_type=AuthTokenType.USER)
+            plaintext_refresh_token = None  # user auth tokens do not have refresh tokens
+        else:
+            # to maintain compatibility with current
+            # code that currently calls create with token= specified
+            if "token" in kwargs:
+                plaintext_token = kwargs["token"]
```

Comment:
There are tests that set tokens that I wouldn't consider valid. For example, their length is way too short. 
I think it would be wise to at least validate length, so I'll get that added in and update all the tests. 

I don't want to change too much behavior in this PR though, so I'll leave that to a future PR -- if you're good with that.

Reply:
Yeah that sounds good to me!